src/gha_logs.rs

```diff
@@ -234,6 +227,93 @@ async fn process_logs(
         }
     };
 
+    let tree_roots = tree_roots.iter().map(|p| regex::escape(p)).join("|");
```

Comment:
Since the logs can be quite large, doing this processing might not be super cheap (the triagebot server isn't very powerful). What about computing all of this only once, and then caching the whole HTML page output? We'd have to check how does its size compare to the size of the raw logs though, it could be much larger than the log size.

I couldn't benchmark it locally because the regex crashed for me.